pennylane_lightning/src/algorithms/AdjointDiff.hpp

```diff
@@ -46,91 +45,53 @@ static constexpr auto getP11() -> std::vector<std::complex<T>> {
     return {ZERO<T>(), ZERO<T>(), ZERO<T>(), ONE<T>()};
 }
 
-template <class T = double, class SVType = Pennylane::StateVector<T>>
+template <class T = double, class SVType>
 void applyGeneratorRX(SVType &sv, const std::vector<size_t> &wires,
                       const bool adj = false) {
-    sv.applyOperation("PauliX", wires, adj);
+    sv.applyPauliX(wires, adj);
 }
 
-template <class T = double, class SVType = Pennylane::StateVector<T>>
+template <class T = double, class SVType>
 void applyGeneratorRY(SVType &sv, const std::vector<size_t> &wires,
                       const bool adj = false) {
-    sv.applyOperation("PauliY", wires, adj);
+    sv.applyPauliY(wires, adj);
 }
 
-template <class T = double, class SVType = Pennylane::StateVector<T>>
+template <class T = double, class SVType>
 void applyGeneratorRZ(SVType &sv, const std::vector<size_t> &wires,
                       const bool adj = false) {
-    sv.applyOperation("PauliZ", wires, adj);
+    sv.applyPauliZ(wires, adj);
 }
 
-template <class T = double, class SVType = Pennylane::StateVector<T>>
+template <class T, class SVType>
 void applyGeneratorPhaseShift(SVType &sv, const std::vector<size_t> &wires,
                               const bool adj = false) {
-    sv.applyOperation(getP11<T>(), wires, adj);
+    sv.applyGeneratorPhaseShift(wires, adj);
```

Comment:
Do we have generator methods as part of the StateVector class now? Would it be better to keep these tied only to the adjoint implementation, as they will not be used outside it?

Reply:
Inside each kernel implementations. `sv.apply...` just call the default kernel implementation. As it is not really a gate operation (it's not unitary), I also think there can be a better option. But as I also want every implementation dependent code inside one class, I want to stick with this now. Of course, it can be refactored sooner or later.